crates/red_knot_python_semantic/src/semantic_index/use_def/symbol_state.rs

```diff
@@ -242,6 +286,53 @@ impl SymbolBindings {
             visibility_constraints: self.visibility_constraints.iter(),
         }
     }
+
+    fn merge(&mut self, b: Self, visibility_constraints: &mut VisibilityConstraints) {
+        let a = std::mem::take(self);
+        self.live_bindings = a.live_bindings.clone();
+        self.live_bindings.union(&b.live_bindings);
+
+        // Invariant: These zips are well-formed since we maintain an invariant that all of our
+        // fields are sets/vecs with the same length.
+        let a = (a.live_bindings.iter())
+            .zip(a.constraints)
+            .zip(a.visibility_constraints);
+        let b = (b.live_bindings.iter())
+            .zip(b.constraints)
+            .zip(b.visibility_constraints);
```

Comment:
Oh, I wasn't aware that we're zipping three iterators here. It could make sense to use `izip` here ;)

Reply:
With the extra set of parentheses around the first iterator, I don't mind how this looks, and tbh that was the only reason I reached for `izip` in the first place!  (I never like it when `rustfmt` takes `A.zip(B).zip(C)` and breaks up the `A` across multiple lines...)